qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,246 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Rough drag pulse calibration experiment."""
+
+from typing import List, Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate, Parameter
+from qiskit.qobj.utils import MeasLevel
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+from qiskit.providers.options import Options
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.data_processing.processor_library import get_to_signal_processor
+from qiskit_experiments.calibration.exceptions import CalibrationError
+from qiskit_experiments.calibration.experiments.drag_analysis import DragCalAnalysis
+
+
+class DragCal(BaseExperiment):
+    r"""An experiment that scans the DRAG parameter to find the optimal value.
+
+    A Derivative Removal by Adiabatic Gate (DRAG) pulse is designed to minimize leakage
+    to a neighbouring transition. It is a standard pulse with an additional derivative
+    component. It is designed to reduce the frequency spectrum of a normal pulse near
+    the :math:`|1\rangle` - :math:`|2\rangle` transition, reducing the chance of leakage
+    to the :math:`|2\rangle` state.
```

Comment:
In my understanding DRAG accounts for collections of two errors. Namely, one that you already described but other one is phase accumulation due to Stark shift (i.e. that is non-real excitation). Seems like only the first part is written in here.

Reply:
You are correct. I added this here: cc9bcca